neo/Prompt/Commands/tests/test_wallet_commands.py

```diff
@@ -240,6 +257,61 @@ def test_wallet_create_address(self):
         self.assertEqual(type(res), UserWallet)
         self.assertEqual(len(res.Addresses), 9)
 
+    def test_wallet_claim_1(self):
+        # test with no wallet
+        args = ['claim']
+        res = CommandWallet().execute(args)
+        self.assertFalse(res)
+
+        self.OpenWallet1()
+
+        # test with max_coins_to_claim non integer
+        args = ['claim', 'abc']
+        claim_tx, relayed = CommandWallet().execute(args)
+        self.assertEqual(claim_tx, None)
+        self.assertFalse(relayed)
+
+        # test with max_coins_to_claim <= 0
+        args = ['claim', '-1']
+        claim_tx, relayed = CommandWallet().execute(args)
+        self.assertEqual(claim_tx, None)
+        self.assertFalse(relayed)
+
+        # test wrong password
+        with patch('neo.Prompt.Commands.Wallet.prompt', side_effect=["wrong"]):
+            args = ['claim']
+            claim_tx, relayed = CommandWallet().execute(args)
+            self.assertEqual(claim_tx, None)
+            self.assertFalse(relayed)
+
+        # test successfull
+        with patch('neo.Prompt.Commands.Wallet.prompt', side_effect=[WalletFixtureTestCase.wallet_1_pass()]):
+            args = ['claim']
+            claim_tx, relayed = CommandWallet().execute(args)
+            self.assertIsInstance(claim_tx, ClaimTransaction)
+            # self.assertTrue(relayed)
+
+        # test nothing to claim anymore
+        # with patch('neo.Prompt.Commands.Wallet.prompt', side_effect=[WalletFixtureTestCase.wallet_1_pass()]):
+        #     args = ['claim']
+        #     claim_tx, relayed = CommandWallet().execute(args)
+        #     self.assertEqual(claim_tx, None)
+        #     self.assertFalse(relayed)
+
+        # test with --from_addr
```

Comment:
How could I test this?

Reply:
I don't know the wallet structure from the top of my head, but what you would do with a `send` (which supports the same argument) is;
- create the tx
- inspect the "vin" list of that tx and make sure that it has 1 input where the `address` field equals to whatever is specified in `--from_addr`

You would need a wallet with at least 2 addresses that hold NEO (assuming that created claimable GAS) or 2 addresses with unclaimed GAS. Then you can test if the claim is actually taken from the right address. I don't know if this is currently supported, never looked at that part.

Reply:
Thank you for the detailed answer.
It seems pretty time consuming to add this test, I would suggest to skip it for now 😅

Reply:
I don't think it's a good idea to skip, because it likely isn't working at this point (like the `max_coins` param isn't). Perhaps @jseagrave21 can help you on it. He's done a similar thing before with `send`/`sendmany`

Reply:
@LysanderGG let me know if I can help.